### PR TITLE
Gate task-board success on delivery evidence

### PR DESCRIPTION
**Pull request by Th0rgal**
## Summary
- reject acknowledgment-plus-future-intent worker messages as failed delivery
- teach workers to emit a structured `DELIVERED:` completion line while preserving legacy terse completion reports
- apply and test the same pure classification rule for live settle and restart zombie sweep

## Validation
- failing-first regression committed separately in 23cbb95d
- full CI gates pending on the final head

---

> [!NOTE]
> **Medium Risk**
> Changes orchestration semantics: false negatives can retry or fail tasks; false positives could still unblock the DAG, but the goal is stricter gating than before.
> 
> **Overview**
> Task-board worker turns no longer count as **Success** unless the final message shows real delivery. **`classify_outcome`** now routes non-blocked, non-error output through **`has_delivery_evidence`**; otherwise the outcome is **Failed** (triggering the existing auto-retry path).
> 
> **`has_delivery_evidence`** treats a last non-empty line starting with **`DELIVERED:`** (including bold variants) as proof, while still accepting legacy terse completion wording via prefix/marker allowlists. It rejects acknowledgements, future intent (`I'll`, `we will`, etc.), partial updates, negated completion phrases, and typography-normalized “OK, sure thing…” replies so dependent tasks are not unblocked on promises alone.
> 
> The **worker contract** now tells workers to end successful turns with a **`DELIVERED:`** summary line. Tests cover the new classifier edge cases and assert **live settle** and **restart zombie sweep** agree on the same rule.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cef6367c79cc93ecae52cf63424465180c63aef2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/api/control/board.rs [hunk 1]
```diff
@@ -131,7 +131,54 @@ pub fn classify_outcome(
...
+        "understood",
+        "got it",
+        "i'll",
+        "i will",
+    ]
+    .iter()
+    .any(|prefix| normalized.starts_with(prefix));
+    let future_intent = [
+        "i'll",
+        "i will",
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs:219]: Fixed in 41be8023: generic `I will` / `I’ll` are no longer acknowledgment prefixes; only explicit start-intent forms are. Added the reported `I will mark this complete: fixed the parser and verified cargo test.` case and assert it remains Success.

File: src/api/control/board.rs [hunk 2]
```diff
@@ -131,7 +131,56 @@ pub fn classify_outcome(
...
+
+/// Require evidence that the worker delivered, rather than merely promising to
+/// start. New workers emit `DELIVERED:`; legacy/free-form completion summaries
+/// remain valid unless the whole short message is shaped like an acknowledgement
+/// plus future intent. Thus terse reports such as "Fixed the parser." still pass.
+fn has_delivery_evidence(output: &str) -> bool {
+    if output.lines().any(|line| {
+        line.trim_start()
+            .to_ascii_uppercase()
+            .starts_with("DELIVERED:")
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in 34d8326a: `has_delivery_evidence` now strips a leading Markdown bold delimiter before matching `DELIVERED:`, mirroring the existing `**BLOCKED` tolerance. Added the exact regression `**DELIVERED:** Fixed the settle path; verified with cargo test.` and assert it classifies as Success.

File: src/api/control/board.rs [hunk 3]
```diff
@@ -131,7 +131,57 @@ pub fn classify_outcome(
...
+        "inspect it",
+        "look into",
+        "work on",
+        "begin",
+        "take a look",
+    ]
+    .iter()
+    .any(|phrase| normalized.contains(phrase));
+
+    !(normalized.chars().count() <= 600 && acknowledgement && future_intent)
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed at exact head 28992a0e. Progress classification now normalizes curly apostrophes and en/em dashes before anchored future-intent matching. Added regressions for both reported bypasses; the earlier outdated findings remain covered by existing tests.

File: src/api/control/board.rs [hunk 4]
```diff
@@ -131,7 +131,57 @@ pub fn classify_outcome(
...
+        "inspect it",
+        "look into",
+        "work on",
+        "begin",
+        "take a look",
+    ]
+    .iter()
+    .any(|phrase| normalized.contains(phrase));
+
+    !(acknowledgement && future_intent)
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed at exact head 28992a0e. Progress classification now normalizes curly apostrophes and en/em dashes before anchored future-intent matching. Added regressions for both reported bypasses; the earlier outdated findings remain covered by existing tests.

File: src/api/control/board.rs [hunk 5]
```diff
@@ -131,7 +131,44 @@ pub fn classify_outcome(
...
+        "get started",
+        "start on",
+        "inspect it",
+        "look into",
+        "work on",
+        "begin",
+        "take a look",
+    ]
+    .iter()
+    .any(|phrase| normalized.contains(phrase));
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed at exact head 28992a0e. Progress classification now normalizes curly apostrophes and en/em dashes before anchored future-intent matching. Added regressions for both reported bypasses; the earlier outdated findings remain covered by existing tests.

File: src/api/control/board.rs [hunk 6]
```diff
@@ -131,7 +131,58 @@ pub fn classify_outcome(
...
+                .map_or(true, |c| c.is_ascii_punctuation() || c.is_whitespace());
+            if boundary {
+                progress = rest
+                    .trim_start_matches(|c: char| c.is_ascii_punctuation() || c.is_whitespace());
+                break;
+            }
+        }
+    }
+    let future_intent = [
+        "i'll",
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs:218]: Fixed at exact head 28992a0e. Progress classification now normalizes curly apostrophes and en/em dashes before anchored future-intent matching. Added regressions for both reported bypasses; the earlier outdated findings remain covered by existing tests.

File: src/api/control/board.rs [hunk 7]
```diff
@@ -131,7 +131,58 @@ pub fn classify_outcome(
...
+    let mut progress = normalized.as_str();
+    for prefix in ["acknowledged", "okay", "ok", "sure", "understood", "got it"] {
+        if let Some(rest) = progress.strip_prefix(prefix) {
+            let boundary = match rest.chars().next() {
+                Some(c) => c.is_ascii_punctuation() || c.is_whitespace(),
+                None => true,
+            };
+            if boundary {
+                progress = rest
+                    .trim_start_matches(|c: char| c.is_ascii_punctuation() || c.is_whitespace());
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed at exact head 28992a0e. Progress classification now normalizes curly apostrophes and en/em dashes before anchored future-intent matching. Added regressions for both reported bypasses; the earlier outdated findings remain covered by existing tests.

File: src/api/control/board.rs [hunk 8]
```diff
@@ -131,7 +131,67 @@ pub fn classify_outcome(
...
+        "inspect it",
+        "look into",
+        "work on",
+        "begin",
+        "take a look",
+    ]
+    .iter()
+    .any(|phrase| progress.starts_with(phrase));
+
+    legacy_completion || !future_intent
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in 57312bd3. After stripping a recognized acknowledgement, empty progress now fails explicitly; regression tests cover both Acknowledged. and OK.

File: src/api/control/board.rs [hunk 9]
```diff
@@ -131,7 +131,65 @@ pub fn classify_outcome(
...
+    // Normalize common typography produced by rich-text clients before
+    // matching progress-only replies. Without this, `I’ll` and an em dash
+    // after an acknowledgement bypass the ASCII-oriented intent checks.
+    let normalized = output
+        .trim()
+        .to_lowercase()
+        .replace(['\u{2018}', '\u{2019}'], "'")
+        .replace(['\u{2013}', '\u{2014}'], "-");
+    let legacy_completion = normalized.starts_with("i will mark this complete:");
+    let mut progress = normalized.as_str();
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in 57312bd3. The legacy completion check now runs after acknowledgement stripping, with a regression for OK, I will mark this complete: ...

File: src/api/control/board.rs [hunk 10]
```diff
@@ -131,7 +131,65 @@ pub fn classify_outcome(
...
+    }
+    let future_intent = [
+        "i'll",
+        "i will",
+        "get started",
+        "start on",
+        "inspect it",
+        "look into",
+        "work on",
+        "begin",
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in 57312bd3. Ambiguous begin/work on prefixes now use word boundaries and completion markers, so completion summaries pass while genuine imperative/future replies still fail. Added regressions for both examples.

File: src/api/control/board.rs [hunk 11]
```diff
@@ -131,7 +131,91 @@ pub fn classify_outcome(
...
+    // Normalize common typography produced by rich-text clients before
+    // matching progress-only replies. Without this, `I’ll` and an em dash
+    // after an acknowledgement bypass the ASCII-oriented intent checks.
+    let normalized = output
+        .trim()
+        .to_lowercase()
+        .replace(['\u{2018}', '\u{2019}'], "'")
+        .replace(['\u{2013}', '\u{2014}'], "-");
+    let mut progress = normalized.as_str();
+    for prefix in ["acknowledged", "okay", "ok", "sure", "understood", "got it"] {
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in 05efb727. Multiword acknowledgements are matched longest-first (including sure thing and sounds good), so the remaining progress begins at the future-intent phrase. Added regressions for both shapes.

File: src/api/control/board.rs [hunk 12]
```diff
@@ -131,7 +131,105 @@ pub fn classify_outcome(
...
+    ] {
+        if let Some(rest) = progress.strip_prefix(prefix) {
+            let boundary = match rest.chars().next() {
+                Some(c) => c.is_ascii_punctuation() || c.is_whitespace(),
+                None => true,
+            };
+            if boundary {
+                progress = rest
+                    .trim_start_matches(|c: char| c.is_ascii_punctuation() || c.is_whitespace());
+                break;
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in bd798ff5. Acknowledgement prefixes are now stripped in a loop, longest-first, so chained acknowledgements expose the actual future-intent text. Added the reported regression.

File: src/api/control/board.rs [hunk 13]
```diff
@@ -131,7 +131,105 @@ pub fn classify_outcome(
...
+            "complete",
+            "completed",
+            "done",
+            "finished",
+            "fixed",
+            "implemented",
+            "verified",
+        ]
+        .iter()
+        .any(|marker| progress.contains(marker));
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in bd798ff5. Negated and unfinished markers are rejected before any positive completion marker is considered; the exact not-complete/continue shape is covered.

File: src/api/control/board.rs [hunk 14]
```diff
@@ -131,7 +131,105 @@ pub fn classify_outcome(
...
+    // `OK, I will mark this complete: ...` retain the documented legacy
+    // completion shape.
+    if progress.starts_with("i will mark this complete:") {
+        return true;
+    }
+
+    let explicit_future_intent = [
+        "i'll",
+        "i will",
+        "get started",
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in bd798ff5. Get Started is treated as intent only in standalone/on/with action forms. A Get Started button fixed summary now passes by positive completion evidence.

File: src/api/control/board.rs [hunk 15]
```diff
@@ -131,7 +131,166 @@ pub fn classify_outcome(
...
+        " were completed",
+        " has been completed",
+        " have been completed",
+        " test passes",
+        " tests pass",
+        " build passes",
+        " build succeeded",
+        " pr merged",
+        " commit pushed",
+        " fixed",
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs:314]: Fixed in bd798ff5. Negated completion forms such as not fixed/not implemented/not complete are rejected before positive markers. Added the exact regression.

File: src/api/control/board.rs [hunk 16]
```diff
@@ -131,7 +131,166 @@ pub fn classify_outcome(
...
+        "analyzed",
+        "changed",
+        "completed",
+        "confirmed",
+        "created",
+        "documented",
+        "done",
+        "finished",
+        "fixed",
+        "found",
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs:288]: Fixed in bd798ff5. First-person remaining-work phrases are detected anywhere in the summary with token boundaries, so Found ...; I will/Ill ... cannot pass via an investigative prefix. Positive legacy evidence is still required.

File: src/api/control/board.rs [hunk 17]
```diff
@@ -131,7 +131,210 @@ pub fn classify_outcome(
...
+/// Require evidence that the worker delivered, rather than merely promising to
+/// start. New workers emit `DELIVERED:`; legacy/free-form completion summaries
+/// remain valid unless the message is shaped like future intent. Thus terse
+/// reports such as "Fixed the parser." still pass.
+fn has_delivery_evidence(output: &str) -> bool {
+    if output.lines().any(|line| {
+        line.trim_start()
+            .trim_start_matches("**")
+            .to_ascii_uppercase()
+            .starts_with("DELIVERED:")
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/control/board.rs]: Fixed in cef6367c. DELIVERED is recognized only on the final non-empty line, and the full turn is still checked for bounded future-intent and unfinished/negated markers before the explicit marker is accepted. The quoted-example shape is covered exactly.
